gopages/main.go

```diff
@@ -114,10 +114,15 @@ var generateMemfsDocsPipe = pipe.New(pipe.Options{}).
 		return args[0].(memfsDocsArgs)
 	}).
 	Append(func(args memfsDocsArgs) (memfsDocsArgs, *git.Repository, error) {
+		var auth *gitHTTP.BasicAuth = nil
+		if args.Flags.GitHubPagesUser != "" || args.Flags.GitHubPagesToken != "" {
+			auth = &gitHTTP.BasicAuth{Username: args.Flags.GitHubPagesUser, Password: args.Flags.GitHubPagesToken}
+		}
```

Comment:
This change looks good and works well locally.

No pressure, but would you mind adding a regression test to cover this scenario?
One easy option might be to extract this bit to a separate func, then add a unit test for both zero auth and GitHub token auth.

Reply:
Added tests as asked. Hope this is good now! :)